python_modules/dagster-test/dagster_test/toys/schedules.py

```diff
@@ -98,7 +98,7 @@ def longitudinal_schedule():
     def longitudinal_config(start, _end):
         return {
             "ops": {
-                op.name: {"config": {"partition": start.to_date_string()}}
```

Comment:
I guess we are even being given some clear signal right here that this could be a breaking change for some folks if we're relying on it ourselves in our example user code..

Is there a world where we use boring datetimes under the hood for everything internal but then send it to a pendulum datetime anywhere its exposed in user code?

Reply:
Alternately we could ship it in 1.8

Reply:
where did you try instantiating it exactly? I thought most of the places where we were acutely feeling the pain here was in dagster system internals like the backfill daemon and AMP daemon, I wouldn't expect it to be needed there since we control all the callsites

Reply:
Like a concrete example would be here (maybe you didn't touch this one?) https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_grpc/impl.py#L312-L319

We would still create a pendulum timestamp anywhere that we are creating some argument that is going to be invoked in user code

I am having a lot of trouble finding the correct place to do that for this specific daily_partitioned_config example though

Reply:
> where did you try instantiating it exactly?

All the yields in `cron_string_iterator`:

```
Total keys 448454
python ./scripts/slowtime.py  7.50s user 0.06s system 99% cpu 7.606 total
```

Reply:
Ok the best I can do is this:
```
Total keys 448469
python ./scripts/slowtime.py  6.18s user 0.07s system 98% cpu 6.315 total
```

With returning pendulum from the iterator. I'll just do that as a starting point.

Reply:
What I had in mind was finding callsites like this

https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/definitions/time_window_partitions.py#L1094-L1117

intercepting the TimeWindow with regular datetimes, and turning it into a TimeWindow with pendulum datetimes before passing it into the user-specified run_config_fn

the tricky part would be identifying all the places where we do that..

Reply:
Another example would be this public context.partition_time_window property that is exposed to user code: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/execution/context/compute.py#L394-L416

Reply:
Yeah I see what you're saying. I think I'm going to aim to land what I have here now and then leave that for future work if somebody wants to pick up the torch from here. It should be straightforward to replace all create_pendulum_timestamp callsites now as well with the ```apply_fold_and_post_transition``` function.